python/revng/scripting/local_daemon_profile.py

```diff
@@ -0,0 +1,74 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import socket
+import subprocess
+from signal import SIGINT
+from time import sleep
+
+import requests
+
+from revng.scripting.projects import DaemonProject
+
+
+class LocalDaemonProject(DaemonProject):
+    def __init__(self, resume_dir: str | None, revng_executable: str | None) -> None:
+        self.port: int = self._get_port()
+        self.daemon_process: subprocess.Popen | None = None
+        self._set_resume_dir(resume_dir)
+        self._set_revng_executable(revng_executable)
+        self.start_daemon()
+
+    def init_client(self, connection_retries: int = 10) -> None:
+        failed_retries = 0
+        while failed_retries <= connection_retries:
+            if self.is_server_running():
+                super().__init__(f"http://127.0.0.1:{self.port}")
```

Comment:
NONONO, call `super()` in `__init__`

Reply:
Can you move `super().__init__` as the first statement? It's what the rest of the codebase does

Reply:
I reworked this part a bit. Now the `DeamonProject` has a `setup_client` method so that we can call that and the end of the `LocalDaemonProject`'s `init`.

Reply:
Why is it needed? Can't the code of `setup_client` be part of `__init__`?

Reply:
Yes, but then the call to `super().__init__` in `LocalDaemonProfile`s `__init__` would need to go to the end. I understood from the previous comment that you want the call to `super` and the begging of the `init`. `Setup_client` must happen after the `deamon` is started.

Reply:
Ok, put the `super().__init__` at the end of `__init__` in `LocalDaemonProfile`